### ISSUE TITLE
RESOLVED: uBlock Failure to Filter a Flash Ad--Request by "Gorhill"

### ISSUE THREAD

**Issue by DoctorX101** [gorhill/uBlock#601]
Requested by "Gorhill" to post this here. ABE effectively frame-blocked flash video ads. If not automatic, you could choose to block it. The site is http://v6.player.abacast.net/2099--safe unless you hate hearing how awesome the Patriots are. . . .

Current Firefox is 31.8.0 ESR--"why no update?"--because the updates have bugs unrelated.

Here is what it looks like WITH ABE still installed:

![001](https://cloud.githubusercontent.com/assets/13725519/9290902/4f822e76-4369-11e5-863e-8147f27a38f6.jpg)

that black "box" is where the LOUD flash video would be. It was selectively blocked with ABE.

Now, taking off ABE and only using uBlock:

![002](https://cloud.githubusercontent.com/assets/13725519/9290901/4f7fad0e-4369-11e5-860d-0b456c196a49.jpg)

"Why you no element block with uBlock?!!"

Here is what happens:

![004](https://cloud.githubusercontent.com/assets/13725519/9290903/68269d40-4369-11e5-927f-5aa9e7bae4c6.jpg)

The ad will not be blocked. No matter how many times you try.

**Comment by gorhill** [comment]
Your link does not work:

> The requested URL /2099--safe was not found on this server.

**Comment by DoctorX101** [comment]
Remove --safe

**Comment by gorhill** [comment]
Creating a filter was tricky because there is a timer on the site which will remove the ad element from the DOM after a short time. I was able to be quick enough to get this:

    v6.player.abacast.net###gateway_player

This will result in the same result as ABE.

Any reason to not simply import your custom filters from ABE?

Side note, there were no ads and no ugly empty container covering the page the first time I visited the site, the site was all clean, and the live streaming worked just fine: I use [medium mode](https://github.com/gorhill/uBlock/wiki/Blocking-mode:-medium-mode), and my Flash plugin is completely disabled.

**Comment by gorhill** [comment]
Alright, I will recapitulate, I am not used to have Flash enabled, and I went sideway here about the original issue. So I did enabled Flash for that site and tried to reproduce your:

> The ad will not be blocked. No matter how many times you try.

While the Flash ad was playing, I clicked "Block element" in the popup menu, and selected the Flash ad, upon which the element picker suggested the same filter as seen in your screenshot. I click _Create_, and the filter was created as expected. So I am not able to reproduce your specific issue.

**Comment by DoctorX101** [comment]
I am not sure how you "simply import your custom filters." This may be obvious but it is not listed in preference or in any of the "toggle" accessed on the icon.

While the filter was created--numerous times--it will not block the ad if you refresh twice, close then restart the browser, et cetera.

I have read that link carefully a number of times; yet, I do not see, for example, how to determine "un-check My filters in the 3rd-party filters pane." It lacks instructions to locate said pane. Perhaps this is obvious to others; nevertheless, the information is not contained in it.

**Comment by lewisje** [comment]
Copy your custom filters from ABE and paste them into uO, because uO can't import filters from other extensions automagically; also, in the "3rd-party filters" tab, "My Filters" should *not* be unchecked, and you should make it checked if it isn't already.

**Comment by DoctorX101** [comment]
Thank you very much for your time. I imagine there is a FAQ that explains that process so I do not bother you with a bunch of  "where is uO . . . is it in my iTunes folder?" I will then do all that, disable ABE, test it, at let you know if that seems to solve the problem.

**Comment by lewisje** [comment]
uO is short for uBlock Origin, sorry, but for the rest, go here, right on the project's main page: https://github.com/gorhill/uBlock#documentation

**Comment by DoctorX101** [comment]
*Just Sitting Here on My Clone* Mayhaps this will help others. I test things on a clone and suggest others do the same so if the @#$#! it up they have not done something permanent.

I found the filters for ABE in the Preferences from Addons page. uBlock Origin is under "Show Dashboard." I saved both my subscribed and "custom" filters as two separate text-files. I then pasted the subscribed in 3rd Party Filters and the custom in the space under My Filters I pasted the text for "custom."

Restarted twice . . . with ABE removed . . . so far so good. 

I suppose there was an easier way of doing it. >.< A crap-load of my filters are "outdated" even after updating but I suspect they are from sites I do not visit.

Thanks for the help!
